### PR TITLE
Fix #8726 - Add permission to data insight

### LINKED ISSUE #8726: Limit reportData endpoin to bots only

### PR DESCRIPTION
**Pull request by TeddyCr**
### Describe your changes :
Fixes #8726 

This PR adds the right logic to handle permission for data insight

#
### Type of change :

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

### CODE REVIEW

File: openmetadata-service/src/test/java/org/openmetadata/service/resources/dataInsight/DataInsightChartResourceTest.java
Comment (TeddyCr) [inline openmetadata-service/src/test/java/org/openmetadata/service/resources/dataInsight/DataInsightChartResourceTest.java:1]: This file was renamed and we added `get_data_insight_data_403` to test user without `viewAll` permission on data insight entity won't see the aggregated data.

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/analytics/ReportDataResource.java [hunk 1]
```diff
@@ -25,20 +25,25 @@
...
 import org.openmetadata.service.security.Authorizer;
+import org.openmetadata.service.security.policyevaluator.OperationContext;
+import org.openmetadata.service.security.policyevaluator.ReportDataContext;
+import org.openmetadata.service.security.policyevaluator.ResourceContextInterface;
 import org.openmetadata.service.util.ResultList;
 
 @Slf4j
-@Path("/v1/analytic/reportData")
+@Path("/v1/analytics/reportData")
 @Api(value = "ReportData collection", tags = "ReportData collection")
```
Comment (sureshms) [inline openmetadata-service/src/main/java/org/openmetadata/service/resources/analytics/ReportDataResource.java:41]: @TeddyCr we avoid using verbs in the method name. Can this be `/v1/analytics/report` or `v1/analytics/data`?

**Review by nahuelverdugo — COMMENTED** [review]
LGTM

File: ingestion/src/metadata/ingestion/ometa/mixins/data_insight_mixin.py [hunk 2]
```diff
@@ -44,7 +44,7 @@ def add_data_insight_report_data(self, record: ReportData) -> ReportData:
             record (ReportData): report data
         """
 
-        resp = self.client.post("/analytic/reportData", record.json())
+        resp = self.client.post("/analytics/dataInsight/data", record.json())
```
Comment (sureshms) [inline ingestion/src/metadata/ingestion/ometa/mixins/data_insight_mixin.py:48]: Can we dataInsight as dataInsights?

**Review by sureshms — APPROVED** [review]

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/analytics/ReportDataResource.java [hunk 3]
```diff
@@ -25,20 +25,25 @@
...
 import org.openmetadata.service.resources.Collection;
 import org.openmetadata.service.security.Authorizer;
+import org.openmetadata.service.security.policyevaluator.OperationContext;
+import org.openmetadata.service.security.policyevaluator.ReportDataContext;
+import org.openmetadata.service.security.policyevaluator.ResourceContextInterface;
 import org.openmetadata.service.util.ResultList;
 
 @Slf4j
-@Path("/v1/analytic/reportData")
+@Path("/v1/analytics/reports")
```
[the remark this answers is not in the document]
Comment (TeddyCr) [reply openmetadata-service/src/main/java/org/openmetadata/service/resources/analytics/ReportDataResource.java]: Sounds good. I'll add description. As far as the tags, I updated based on what you mentioned during our discussion.

**Review by sureshms — APPROVED** [review]

**Pull request merged**